### ISSUE TITLE
Can't open Microsoft Teams - blank screen

### ISSUE THREAD

**Issue by sant123** [IsmaelMartinez/teams-for-linux#171]
**Describe the bug**
Trying to open Microsoft Teams leaves a blank page.

**To Reproduce**
1. Open Microsoft Teams
2. That's it, a blank page should be shown (the error)

**Expected behavior**
Load normally

**Screenshots**
![image](https://user-images.githubusercontent.com/7959437/61073245-9769c480-a3da-11e9-8cbf-79542f19a01d.png)

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04.2
 - Installation package deb
 - Version teams-for-linux_0.4.0_amd64.deb

**Additional context**
Clicking on debug option I get this:

![image](https://user-images.githubusercontent.com/7959437/61073379-e6175e80-a3da-11e9-8c81-d7850813f6af.png)

Any idea of what's going on? Thanks.

**Comment by sant123** [comment]
Also when cleaning config "~/.config/Teams for Linux" it solves the issue for a while, then you will encounter this again.

**Comment by IsmaelMartinez** [comment]
I have seen this before but if you click ctrl+R it should just do the job.

Feel free to investigate the root cause, if you can/wish. We really appreciate contributions.

**Comment by IsmaelMartinez** [comment]
Also, 0.4.0 is not out yet as deb package. Only out to snap edge at the moment. Can you confirm deb package version or if it is from snap edge/source? Thanks

**Comment by sant123** [comment]
Thanks for your answer @IsmaelMartinez. I tried Ctrl+R too but didn't work. What I did was sign out and sign in again but that only works once. 

I built the deb installer with `yarn dist:linux deb` command and got these:

![image](https://user-images.githubusercontent.com/7959437/61091552-3953d600-a408-11e9-9a7a-50e876455de9.png)

**Comment by IsmaelMartinez** [comment]
Have you tried the option —disable-gpu.

The windows client seems to have the same issue. For reference:
https://techcommunity.microsoft.com/t5/Microsoft-Teams/Blank-screen-when-opening-Teams-app-on-windows-10/m-p/32356?advanced=false&collapse_discussion=true&q=Teams%20app%20white&search_type=thread

Also, are you company or personal profiles?

**Comment by g4borg** [comment]
I experience the same bug in 0.1.17, upgraded to 0.3.0, and still have that bug.
Same javascript errors. Sometimes refresh helps, sometimes not.

Wow, the microsoft report is from 2016

**Comment by IsmaelMartinez** [comment]
I think is more MS issue, I sometimes need to click a couple of times Ctrl+R but it's pretty sporadic.

It could be possible to log for that log in the console and trigger a refresh, but that might create a infinite loop/kaos.

**Comment by WellingtonCosta** [comment]
+1

**Comment by sant123** [comment]
> Can you explain more about Ctrl+R not working? Have you tried clicking on the “refresh” option in the tab menu?
> 
> Another option is to try —disable-gpu?
> 
> teams-for-linux --disable-gpu
> 
> The windows client seems to have the same issue. For reference:
> https://techcommunity.microsoft.com/t5/Microsoft-Teams/Blank-screen-when-opening-Teams-app-on-windows-10/m-p/32356?advanced=false&collapse_discussion=true&q=Teams%20app%20white&search_type=thread
> 
> Also, are you company or personal profiles? Proxy’s can be also creating this issue.
> 
> Last but not least, can you check snap edge, revert to 0.3.0 or drop me and email and I send you a deb package. It might be an environment issue.

What I mean is hitting Ctrl + R refresh the page but the blank page still persists. Also tried with the "refresh" option in the tab menu and still the same.

This is what I get with the `--disable-gpu` option:

![image](https://user-images.githubusercontent.com/7959437/61135105-9c844d80-a486-11e9-82ed-27821170d7b9.png)

My profile is from a company and was working fine 2 days ago with 0.3.0 version, when issue appeared I upgraded to 0.4.0 to see what happens and happens the same :sweat_smile:

**Comment by IsmaelMartinez** [comment]
have you tried removing the cache? 

https://github.com/IsmaelMartinez/teams-for-linux/blob/develop/KNOWN_ISSUES.md#no-history

Sometimes it does the trick.

**Comment by sant123** [comment]
Yes, but that's my workaround to make it work once. If you do a refresh, you will see the blank page again.

**Comment by IsmaelMartinez** [comment]
do you have the same issue with the workaround in 0.3.0? I know this process might be a bit painful, but will help nail down if the issue is now worse for you or is just something introduced since 0.3.0.

Funnily enough, I use to have to click Ctrl+R every morning but I don't need to do it anymore. MS might have changed something underneath, or your company network.  Apologies for not been that helpful at the moment.

**Comment by sant123** [comment]
Yes I have the same issue with the same workaround. Don't worry, I saw the code last night but understood nothing :rofl: I tried to figure out the issue but seems MS Teams issue, in fact is done in Angular.js If I get an idea of what's going on I'll post it here.

**Comment by g4borg** [comment]
tried all 0.1.16, 0.1.17, 0.3.0 and deleting the .config folder each time, but at second time it always happens, even in different internet scenarios. 
Relogging works as workaround, either by waiting for the timeout site of teams, or deleting the config folder. Albeit, our SSO logout does not bring you back to the login screen unfortunately, so usually I delete the config.

The main difference is now, that it always happens now for me, and it will not recover with refresh at all, on all versions.

> Error: Action with name notificationsService_startup already exists

i think this angular error basicly happens only on reload, not on first login. And it does not even work after a refresh once you logged in successfully.

> The Content-Security-Policy directive 'prefetch-src' is implemented behind a flag which is currently disabled

Not sure if this might be related. at least i can try out all fixes because it is always reliably does not work.

**Comment by lgringo** [comment]
Hi !
First, thanks for your work. It's very convenient to have a Linux client for Teams.

I don't know if it can help, but deleting the folder : 
.config/Teams for Linux/Partitions/teams-4-linux/Service Worker

resolves the problem (@sant123 said deleting .config/Teams for Linux solves the problem, but you have to do a full login, not when you delete only Service Worker)

**Comment by g4borg** [comment]
confirmed, that does the trick for me as well.

**Comment by IsmaelMartinez** [comment]
does any of you all can run the app from source? My suspect now is the cache so it might be as easy as changing this line https://github.com/IsmaelMartinez/teams-for-linux/blob/develop/app/index.js#L74 to:
```
window.loadURL(config.url, { extraHeaders: 'pragma: no-cache\n' });
```

from https://electronjs.org/docs/api/web-contents#contentsloadurlurl-options

I do really need to do some refactoring to that index.js file...

**Comment by sant123** [comment]
I executed the app from source with that change but occurs the same thing:

![image](https://user-images.githubusercontent.com/7959437/61307660-58f65000-a7b4-11e9-96c6-82cbc8f8a5a3.png)

**Comment by IsmaelMartinez** [comment]
Well, worth a try. I hope I start facing the issue should I can then fix it!

**Comment by EdRoxter** [comment]
> Hi !
> First, thanks for your work. It's very convenient to have a Linux client for Teams.
> 
> I don't know if it can help, but deleting the folder :
> .config/Teams for Linux/Partitions/teams-4-linux/Service Worker
> 
> resolves the problem (@sant123 said deleting .config/Teams for Linux solves the problem, but you have to do a full login, not when you delete only Service Worker)

Can confirm, need to do this before every start of the application though. I configured a "rm -rf ~/.config/Teams for Linux/Partitions/teams-4-linux/Service Worker/*" as user login script for the time being.

**Comment by WellingtonCosta** [comment]
I deleted `~/.config/Teams for Linux/` folder and it didn't recreated after installing Teams for Linux again. Any idea?

**Comment by g4borg** [comment]
you did try to start teams as well as just installing it? the binary creates it that home folder.
(might be a stupid answer, and nothing to do with your problem)

**Comment by WellingtonCosta** [comment]
I uninstalled and installed multiple times and this folder was not created again.

**Comment by sant123** [comment]
@WellingtonCosta you need to open Microsoft Teams in order to re-create that folder, and yes @EdRoxter you need to delete "Service Worker" folder before opening the application.

**Comment by WellingtonCosta** [comment]
@sant123 I did it, but the folder wasn't recreated...

**Comment by WellingtonCosta** [comment]
I'll try to reconfigure the packge.

**Comment by IsmaelMartinez** [comment]
check if is not been created under `/home/$HOME/snap/teams-4-linux`

**Comment by WellingtonCosta** [comment]
@IsmaelMartinez you're goddamn right!

**Comment by WellingtonCosta** [comment]
@EdRoxter it worked like a charm! Thank you!

**Comment by IsmaelMartinez** [comment]
It will be absolutely amazing if one of you can add it to the known issues section. As other people might encounter the problem. Thanks!!

**Comment by gstorer** [comment]
Also had this issue. It is also resolved by deleting the Service Worker folder from my config directory before start up.

**Comment by vhakulinen** [comment]
Can confirm, deleting service worker folder resolved this issue for me too.

**Comment by IsmaelMartinez** [comment]
> It will be absolutely amazing if one of you can add it to the known issues section. As other people might encounter the problem. Thanks!!

Any volunteers? It should not take you long to put a sentence in it... and I will really appreciate it.

**Comment by sant123** [comment]
Hi @IsmaelMartinez, take a look at https://github.com/IsmaelMartinez/teams-for-linux/pull/178

**Comment by IsmaelMartinez** [comment]
Thanks @sant123 ! Closing this as there is a documented workaround. (I might amend it a bit before the full release but it does save me a lot of time!)

**Comment by sant123** [comment]
Just want to share a bash script to automate the process of closing, opening, removing Service Worker folder:

``` bash
teamsInstances=$(pgrep teams-for-linux)
serviceWorkerFolder="$HOME/.config/Teams for Linux/Partitions/teams-4-linux/Service Worker"

teamsNotRunning="Teams not running, just deleting Service Worker."
serviceWorkerNotFound="Service Worker folder does not exist. Open Teams as normal."
teamsWasRunning="Teams was running, killing and restarting."

if [ -z "$teamsInstances" ]; then
	if [ -d "$serviceWorkerFolder" ]; then
		echo "$teamsNotRunning"
		rm -r "$serviceWorkerFolder"
	else
		echo "$serviceWorkerNotFound"
	fi
else
	echo "$teamsWasRunning"
	kill -9 $teamsInstances
	rm -r "$serviceWorkerFolder"
	teams-for-linux&
fi
```
![image](https://user-images.githubusercontent.com/7959437/61671336-bacf2200-acac-11e9-8a6b-7d6a1a97a4eb.png)

**Comment by Mairu** [comment]
But why not removing this on start automatically when it is a known issue? I really have to delete it every time.

**Comment by IsmaelMartinez** [comment]
I am sorry to hear about that. Feel free to create a PR. It will need to work in all packaged versions.

**Comment by fosspill** [comment]
I seem to be having this worse than most others here. Mine will not load properly unless I delete everything under ".config/teams-for-linux/Partitions/teams-4-linux/". This means I have to reauthenticate about 10 times a workday.

Using the normal web interface works fine, of course, but it would be lovely to have a proper fix for this issue.

**Comment by IsmaelMartinez** [comment]
Have you tried other package types?

**Comment by Phitherek** [comment]
For me, clearing everything under 'Application Cache' works. 'Service Worker' cleaning seemed to work a while ago, but no longer does. (DEB package, v. 0.5.0)

**Comment by leoperria** [comment]
thanks @Phitherek: confirmed that removing "Service Worker" didn't work but removing everything under "Application Cache" worked

**Comment by IsmaelMartinez** [comment]
maybe then we can just clear the cache of the application on start... 
https://github.com/electron/electron/blob/master/docs/api/session.md#sesclearcache

Again, I don't have the issue (or pretty randomly and it resolves itself) but I am going to reopen this as there might be a way around. Anyone interested on picking it up, great! I got little time and spend most of it administering this (and trying to move to electron 6)

**Comment by rojaro** [comment]
Why not leave the ticket open then? Maybe add a `help wanted` label?

**Comment by IsmaelMartinez** [comment]
seems like I didn't click the reopen and comment button... appologies but yeah, the implementation shoudl be fairly easy so adding the `help wanted` and `good first issue`

**Comment by IsmaelMartinez** [comment]
I think chrome is playing with me... had to add the tags twice. time to go home and relax.

**Comment by petr-nehez** [comment]
Same issue @ me :disappointed:

**Comment by abelsromero** [comment]
I started experiencing this issue 2 days ago after a couple of months with no issues, and the only thing that fixes it is deleting `Service Worker` and `Application Cache`.

**Comment by alfsch** [comment]
Same issue on my Machine (Ubuntu 18.0.4 hwe Kernel 5.0) with teams for linux 0.6.0.
For me it's enough to execute rm -Rf "$HOME/.config/teams-for-linux/Partitions/teams-4-linux/Application Cache" before launching to workaround the blank screen.

**Comment by mweirauch** [comment]
t4l 0.6.0, Fedora 30: can confirm, removing "Application Cache" does the trick. Better then deleting "Service Worker".

**Comment by seboss666** [comment]
Hello,

I've installed it this week on my fresh Manjaro KDE installation on work computer (from AUR : https://aur.archlinux.org/packages/teams-for-linux, still in 0.4.0). I've experienced this blank page frustration after a computer reboot, I confirm that 'Application Cache' folder removal is effective while 'Service worker' was not. You may update the wiki with this fresh information :)

**Comment by petr-nehez** [comment]
For me on MX-Linux removing of `Application Cache` is not enough, I have to remove `Service Worker` as well.

**Comment by Phitherek** [comment]
Ok, this is just frustrating now.
The workaround sometimes does work and sometimes doesn't.
I also added 'Cache' to the removal list, seemed like it works but yet sometimes it does not.

**Comment by Phitherek** [comment]
Never mind my previous comment, I made a stupid error in my launch wrapper script :D. Removing 'Service Worker' and 'Application Cache' contents works as a workaround.

**Comment by tomas-lipensky** [comment]
It started to happen after update of Ubuntu 18.04. Before no issues. After blank page and deletion of '~/.config/teams-for-linux/Partitions/teams-4-linux/Application\ Cache/' is required.

I have upgraded:
azure-cli base-files dkms dotnet-sdk-2.1 firefox firefox-locale-cs firefox-locale-en gir1.2-gst-plugins-base-1.0 gir1.2-gstreamer-1.0 gir1.2-gtk-3.0 google-chrome-stable gstreamer1.0-alsa gstreamer1.0-gl gstreamer1.0-gtk3 gstreamer1.0-libav gstreamer1.0-plugins-bad gstreamer1.0-plugins-base gstreamer1.0-plugins-base:i386 gstreamer1.0-plugins-base-apps gstreamer1.0-plugins-good gstreamer1.0-plugins-ugly gstreamer1.0-pulseaudio gstreamer1.0-tools gstreamer1.0-x gtk-update-icon-cache language-pack-en language-pack-gnome-en ldap-utils libcephfs2 libegl-mesa0 libegl1-mesa libfreerdp-client2-2 libfreerdp2-2
libgail-3-0 libgbm1 libgl1-mesa-dri libgl1-mesa-dri:i386 libgl1-mesa-glx libgl1-mesa-glx:i386 libglapi-mesa libglapi-mesa:i386 libglx-mesa0 libglx-mesa0:i386 libgstreamer-gl1.0-0 libgstreamer-plugins-bad1.0-0 libgstreamer-plugins-base1.0-0 libgstreamer-plugins-base1.0-0:i386 libgstreamer-plugins-good1.0-0 libgstreamer1.0-0 libgstreamer1.0-0:i386 libgtk-3-0 libgtk-3-bin libgtk-3-common libinput-bin libinput10 libldap-2.4-2 libldap-2.4-2:i386 libldap-common libmysqlclient20 libnginx-mod-http-geoip libnginx-mod-http-image-filter libnginx-mod-http-xslt-filter libnginx-mod-mail libnginx-mod-rtmp libnginx-mod-stream libnss-myhostname libnss-systemd libosmesa6 libosmesa6:i386 libpam-systemd librados2 libsox-fmt-alsa libsox-fmt-base libsox3 libsystemd-dev libsystemd0 libsystemd0:i386 libudev1 libudev1:i386 libwayland-egl1-mesa libwinpr2-2 libxatracker2  linux-generic-hwe-18.04 gnome-software-plugin-flatpak openjdk-11-jre openjdk-11-jre-headless openjdk-8-jre openjdk-8-jre-headless patch systemd-sysv ubuntu-desktop ubuntu-minimal ubuntu-standard usb-creator-common usb-creator-gtk linux-libc-dev mesa-va-drivers mesa-vdpau-drivers

and installed: pulseeffects pulseaudio-equalizer

Probably one of the following can be responsible for the behavior.

**Comment by kamax7C0** [comment]
Experiencing this on every app start on Ubuntu 18.04 LTS. 
Running "rm -rf ~/.config/teams-for-linux/Partitions/teams-4-linux/Application\ Cache"  before the app start solves it.

**Comment by petr-nehez** [comment]
@tomas-lipensky @kamax7C0 It has nothing to do with Ubuntu.
I am on MX-Linux (which is based on Debian) and have not updated any packages recently and still I get the same issue.

It must be something MS has changed...

**Comment by IsmaelMartinez** [comment]
Added some stuff to clear the cache. Merging now in develop so it should land in snap edge channel as 0.6.2. Can you snap people test it? I haven't find any issues but I didn't had any issues before (so difficult to test)

**Comment by lgringo** [comment]
Hello, looks likes folder Service Worker does not exist anymore. At least, it's not re-created. I have to delete Application Cache (and only this one now).
Running 0.6.0

**Comment by kamax7C0** [comment]
> Added some stuff to clear the cache. Merging now in develop so it should land in snap edge channel as 0.6.2. Can you snap people test it? I haven't find any issues but I didn't had any issues before (so difficult to test)

installed the --edge version via Snap, same blank screen appears on refresh or new start. 
Console shows: 
`SettingsService: Refreshing settings. Array(1)
msteams-web-bridge.min.js:1 SettingsService: Same key already exists in lower case lookup table,             existing value: enableHidControlWhileSfBisRunning,             duplicate: key: enableHidControlWhileSfBIsRunning, value: false Array(1)
warn @ msteams-web-bridge.min.js:1
msteams-web-bridge.min.js:1 SettingsService: Refreshing settings. Array(1)
msteams-web-bridge.min.js:1 SettingsService: Same key already exists in lower case lookup table,             existing value: enableHidControlWhileSfBisRunning,             duplicate: key: enableHidControlWhileSfBIsRunning, value: false Array(1)
warn @ msteams-web-bridge.min.js:1
angular.min.js:35 Uncaught IPC object is null
angular.min.js:112 Error: Action with name notificationsService_startup already exists
    at r.registerAction (https://statics.teams.microsoft.com/hashedjs/3.1-app.min-47228058.js:2008:5774)
    at new n (https://statics.teams.microsoft.com/hashedjs/4-app.desktop.min-b52e7579.js:9:1263)
    at Object.instantiate (https://statics.teams.microsoft.com/hashedjs/0-angular-jquery.min-8200dd6f.js:37:491)
    at Object.<anonymous> (https://statics.teams.microsoft.com/hashedjs/0-angular-jquery.min-8200dd6f.js:38:260)
    at Object.invoke (https://statics.teams.microsoft.com/hashedjs/0-angular-jquery.min-8200dd6f.js:37:390)
    at Object.$get (https://statics.teams.microsoft.com/hashedjs/0-angular-jquery.min-8200dd6f.js:35:71)
    at Object.invoke (https://statics.teams.microsoft.com/hashedjs/0-angular-jquery.min-8200dd6f.js:37:390)
    at https://statics.teams.microsoft.com/hashedjs/0-angular-jquery.min-8200dd6f.js:39:175
    at d (https://statics.teams.microsoft.com/hashedjs/0-angular-jquery.min-8200dd6f.js:36:369)
    at e (https://statics.teams.microsoft.com/hashedjs/0-angular-jquery.min-8200dd6f.js:37:96) undefined
(anonymous) @ angular.min.js:112
The Content-Security-Policy directive 'prefetch-src' is implemented behind a flag which is currently disabled.

_#/conversations/General?threadId=19:ae58acc1466447a0ad23f509f7823ac0@thread.skype&ctx=channel:1 The Content-Security-Policy directive 'prefetch-src' is implemented behind a flag which is currently disabled.

_:356 index.html: remove startup cache listeners
msteams-web-ng1-services.min.js:1 Cleaned the join location
serviceworker.js:362 Error: service-worker unable to find response in cache for : https://statics.teams.microsoft.com/hashedcss/stylesheets.theme-default.min-d45ed988.css
    at serviceworker.js:358` 

oh and now I don't know where to find "Application Cache" in order to delete it ...

**Comment by lgringo** [comment]
Looks like "blank screen" appears only on second start. But if you want for a very long time, you get connected after all ... and next restart works flawlessly (normal speed).

Step to reproduce : 

1. Delete ~/.config/teams-for-linux (rm -rf ~/.config/teams-for-linux/)
2. Start teams-for-linux
3. Login
4. You may have to choose "Use Wep Application"
5. Quit teams-for-linux (right click on systray)
6. Restart teams-for-linux
7. Blank page on step "Initialisation ..." 
8a. Wait
8b. Wait
8c. Wait
8d. Wait
9. After about 2 minutes, it gets connected
10. Quit teams-for-linux (right click on systray)
11. Restart teams-for-linux
12. Only 5 seconds to get connected ... no more blank screen

Can someone confirm ?

**Comment by apetrelli** [comment]
> Looks like "blank screen" appears only on second start. But if you wait for a very long time, you get connected after all ... and next restart works flawlessly (normal speed).
> 
> Step to reproduce :
> ...
> Can someone confirm ?
> 
> (Same thing running 0.6.0, or using 0.6.2 from sources)

Mmm... I tried with 0.6.0 version and no, it stucks into "Initialisation..."  so I cannot go after step 8d (or 8z...).
Anyway deleting Service Worker does not help, I have to delete the whole teams-for-linux directory.

**Comment by petr-nehez** [comment]
This MS Teams product sucks :disappointed: (unfortunately I have to use it and thus I really appreciate `teams-for-linux`)
After few hours I got a screen with MS' issue and after pressing their refresh button it got stuck on the same blank screen...

**Comment by apetrelli** [comment]
> @tomas-lipensky @kamax7C0 It has nothing to do with Ubuntu.
> I am on MX-Linux (which is based on Debian) and have not updated any packages recently and still I get the same issue.
> 
> It must be something MS has changed...

FWIW yesterday I logged in inside the "normal" Teams webapp and a message said "The app has been updated". It appeared only once, so I have no screenshot, sorry.

**Comment by jmayday** [comment]
Looks like I have same problem, but to make sure I would have to open debug window - how to do it?

**Comment by IsmaelMartinez** [comment]
You can click Ctrl+D , or click In the task bar icon and "debug"

**Comment by jmayday** [comment]
OK, so it's same issue here:
```
Error: Action with name notificationsService_startup already exists
```

**Comment by IsmaelMartinez** [comment]
> Hello,
> 
> I've installed it this week on my fresh Manjaro KDE installation on work computer (from AUR : https://aur.archlinux.org/packages/teams-for-linux, still in 0.4.0). I've experienced this blank page frustration after a computer reboot, I confirm that 'Application Cache' folder removal is effective while 'Service worker' was not. You may update the wiki with this fresh information :)

Please do update the know_issues. I got plenty of my plate currently so got no time for much than reply to this stuff, prepare releases and do normal life! I do appreciate contributions when they are possible.

This issue seems to be fairly random affecting quite a few people but at different levels. It also affects MS official electron app for windows, so I suspect is something more related to how electron stores those files than anything. Maybe they get corrupted on close or something like that. 

Can you guys that are having the worse of it try doing a full clean? Meaning, remove the config folder and the app (uninstall) and reinstall again. Maybe that can help. 

Apologies again but the number of users is starting to be unmanageable for me alone. Any help is more than welcomed.

**Comment by brutog** [comment]
+1

Had no issues a while ago. This started just recently (past 2 weeks maybe).

**Comment by petr-nehez** [comment]
I removed the config folder and installed 0.6.2 -> opened the app -> showed debug -> restarted -> and voila the blank screen is here...
The console log shows still the same error: `Action with name notificationsService_startup already exists`.

This indicates to me a generic problem within the Teams' code.

**Comment by lkrms** [comment]
> +1
> 
> Had no issues a while ago. This started just recently (past 2 weeks maybe).
> 
> Anyone else having this problem use Single Sign On? I bounce around from white screen to white screen before I make it teams. Wondering if maybe that is a commonality.

I'm experiencing this in a SSO environment too. I don't have a non-SSO Teams account handy but when I have spare time (ha!), will see if I can figure out a way of testing that for comparison. Unless someone else here gets to it first :)

**Comment by IsmaelMartinez** [comment]
> I removed the config folder and installed 0.6.2 -> opened the app -> showed debug -> restarted -> and voila the blank screen is here...
> The console log shows still the same error: `Action with name notificationsService_startup already exists`.
> 
> This indicates to me a generic problem within the Teams' code.

For clarity, Did you uninstall all previous versions of teams for linux? Also, Did you close the app "quickly" or let it run for a bit? I am starting to suspect corruption in some of the downloaded files from the web.

**Comment by petr-nehez** [comment]
I did not uninstall all previous versions - for sure I will do it now.

And when I am checking now my previous comment I have to correct it:
_...-> opened the app -> showed debug -> refreshed -> blank screen_

I did not restart the app, I only executed "Refresh" command from the tray menu.

**Comment by apetrelli** [comment]
FWIW, I have an old installation on my other PC, version 0.1.9 and it seems to work well, aside from the fact that desktop sharing is not active. Sorry I cannot test more in these days, I will be out starting tomorrow.

**Comment by petr-nehez** [comment]
So full uninstall did not help - this is an error after "Refresh" command:

![image](https://user-images.githubusercontent.com/1265990/62759567-f413da00-ba81-11e9-8dd9-b970320a5b64.png)

**Comment by IsmaelMartinez** [comment]
Maybe we need to do a flush on closing the app

https://electronjs.org/docs/api/session#sesflushstoragedata

https://github.com/IsmaelMartinez/teams-for-linux/blob/599e140b3ca8a8e549b6fdd56c8075b14503c0bf/app/mainAppWindow/index.js#L48

And maybe even adding a https://electronjs.org/docs/api/app#appquit to that same area.

Probably good to also add extra logs for the application events.

**Comment by apetrelli** [comment]
Again, FWIW in version 0.1.9 I see totally different logs.

![Screenshot_20190809_090615-cropped](https://user-images.githubusercontent.com/966981/62760917-8669ad00-ba85-11e9-9688-e34487609873.png)

**Comment by plafue** [comment]
FWIW: I had this problem with the snap `stable` version... had to remove it via `sudo snap remove teams-for-linux`  and then manually delete the `./snap/teams-for-linux` folder. After that installed the `stable` channel version again, started the application, logged in,  and got past the blank screen.

I exited the application and opened it again and got the same `Uncaught IPC object is null` and `Action with name notificationsService_startup already exists` errors as shown in https://github.com/IsmaelMartinez/teams-for-linux/issues/171#issuecomment-519800293.

**Comment by prazooo** [comment]
Had the same issue from yesterday, and following the instructions above I tried to delete the cache and try again. First of all I did not have the folder "~/home/.cache/teams-for-linux". Instead I had a folder named "~/home/.cache/teams-ubuntu-enhanced". Anyway i deleted this folder, uninstalled the existing teams application (0.6.0) and reinstalled the latest edge version (0.6.2). The application started successfully after logging in. 

When I checked the cache folder again, I dont see any teams related cache folder created at all. (Means no cache??) And If I refresh, again it gets stuck in "Initializing" phase"

**Comment by skyghis** [comment]
@prazooo the folder is in `.config` not in `.cache`.

You should delete `.config/teams-for-linux/Partitions/teams-4-linux/Application Cache`.

**Comment by prazooo** [comment]
Im sorry. I mistyped. It should be .config NOT .cache.
I still dont see a folder called "config/teams-for-linux". So I have nothing to delete now. :D

**Comment by plafue** [comment]
For snap installations the `Application Cache` folder can be found under `~/snap/teams-for-linux/current/.config/teams-for-linux/Partitions/teams-4-linux/`

**Comment by prazooo** [comment]
> For snap installations the `Application Cache` folder can be found under `~/snap/teams-for-linux/current/.config/teams-for-linux/Partitions/teams-4-linux/`

Great Thanks. Its there.

**Comment by mweirauch** [comment]
t4l: 0.6.2; Fedora30
Started out with a fresh RPM install and `.config/teams-for-linux` cleaned.
1. Started up, logged in, worked.
1. Quit the app. No `Application Cache` folder.
1. Startup, all fine.
1. Subsequent starts failed. CTRL+Rs didn't help.

My observation/testing:
* always deleted the `Application Cache` folder before the app was started
* most of the times, the `Application Cache` folder **is not** created on shutdown
* on every succesful start (aka no blank screen) of the application, **no** `Application Cache` folder is present during runtime
* on failed starts (aka blank screen), the `Application Cache` folder is **created during start** of the application

So it seems to me, despite the fix of removing the `Aplication Cache` folder before application start, sometimes that folder is created during start and these times it leads to the "blank screen" issue.
HTH.

**Comment by IsmaelMartinez** [comment]
@mweirauch  , I do have an application cache in my deb app and it works fine. 

I added a few more things to the snap --edge (0.6.3) that should provide more logs about the application closing, etc.

I also added the following to clear the appcache on startup
```
window.webContents.session.clearStorageData({ storages: ['appcache']} ,()=> console.log("appcache cleared"));
```

It doesn't remove the application cache folder but might be the right direction without been too aggressive. 

I manage to get a couple of times stuck in the white screen (manage to get snap running!!) but after adding that clearStorageData I "only" need to Ctrl+Q (quit the app) and start it again. 

If this proves to be the case across the board we can implement something around that (like restart the app if we find that IPC message and/or wait for the application cache to be clean before loading anything)

**Comment by kamax7C0** [comment]
> @mweirauch , I do have an application cache in my deb app and it works fine.
> 
> I added a few more things to the snap --edge (0.6.3) that should provide more logs about the application closing, etc.
> 
> I also added the following to clear the appcache on startup
> 
> ```
> window.webContents.session.clearStorageData({ storages: ['appcache']} ,()=> console.log("appcache cleared"));
> ```
> 
> It doesn't remove the application cache folder but might be the right direction without been too aggressive.
> 
> I manage to get a couple of times stuck in the white screen (manage to get snap running!!) but after adding that clearStorageData I "only" need to Ctrl+Q (quit the app) and start it again.
> 
> If this proves to be the case across the board we can implement something around that (like restart the app if we find that IPC message and/or wait for the application cache to be clean before loading anything)

Refreshed to 0.6.3 : 

now the behavior has changed a bit in a way that: 
- Say I start the app ok 
- Quit. Start again - stuck on a white screen with the same JS error [1] 
- Quit. Start again - it works. 
- Quit. Start again - stuck. 
- Quit, start - works 
- etc ... it loops. So it's probably cleaning up the cache every second start. 

[1] 
![Screenshot from 2019-08-09 12-11-46](https://user-images.githubusercontent.com/17438994/62772283-6941d800-ba9f-11e9-9e5f-655eaa694ee3.png)

**Comment by IsmaelMartinez** [comment]
got no time now, maybe tomorrow, but there is a way there... maybe we need to wait for the clean before loading it. Apologies but I need to park it for just now.

**Comment by mivanovaxway** [comment]
Does adding ```window.webContents.session.clearStorageData({ storages: ['appcache']} ,()=> console.log("appcache cleared"));``` fix the issue for you? Seems it does for me.

**Comment by davidcolacilli** [comment]
Clearing storage did the job, for now

**Comment by abelsromero** [comment]
> got no time now, maybe tomorrow,

Take your time 👍 Current scenario is a nuisance but I think deleting the cache works for most people and is not such a big thing.

**Comment by Lomanic** [comment]
https://github.com/IsmaelMartinez/teams-for-linux/issues/171#issuecomment-519933176

You can keep the "Cookies" checkbox unchecked so you don't have to log in every time.

**Comment by IsmaelMartinez** [comment]
> > @mweirauch , I do have an application cache in my deb app and it works fine.
> > I added a few more things to the snap --edge (0.6.3) that should provide more logs about the application closing, etc.
> > I also added the following to clear the appcache on startup
> > ```
> > window.webContents.session.clearStorageData({ storages: ['appcache']} ,()=> console.log("appcache cleared"));
> > ```
> > 
> > 
> > It doesn't remove the application cache folder but might be the right direction without been too aggressive.
> > I manage to get a couple of times stuck in the white screen (manage to get snap running!!) but after adding that clearStorageData I "only" need to Ctrl+Q (quit the app) and start it again.
> > If this proves to be the case across the board we can implement something around that (like restart the app if we find that IPC message and/or wait for the application cache to be clean before loading anything)
> 
> Refreshed to 0.6.3 :
> 
> now the behavior has changed a bit in a way that:
> 
> * Say I start the app ok
> * Quit. Start again - stuck on a white screen with the same JS error [1]
> * Quit. Start again - it works.
> * Quit. Start again - stuck.
> * Quit, start - works
> * etc ... it loops. So it's probably cleaning up the cache every second start.
> 
> [1]
> ![Screenshot from 2019-08-09 12-11-46](https://user-images.githubusercontent.com/17438994/62772283-6941d800-ba9f-11e9-9e5f-655eaa694ee3.png)

Hi, can you check with the latest (0.6.4)? I am using that async/await magic now to wait for the cache to be clean.

**Comment by seasurfpete** [comment]
Hi guys, I too started getting this problem a few days ago on v0.3.0 on Solus OS. I've tried all the releases with the same issue. the only way i can get it to load is to blow away the .config directory and i'll work just fine (we have login via a corp redirect thing too but that all works)

first fresh load always works, then it'll just hang at the js error. i did notice that of all the MS sites, teams is VERY time dependant on the login after having it wrong and ended up in a redirect loop. was hoping that was the issue but no :(

thanks for all your good work on this.

**Comment by basaas** [comment]
We ran into this issue with MS Teams on our Electron app (Mac & Windows) too.
What worked for us was overwriting the serviceworker.register method via [Webview preload script](https://electronjs.org/docs/api/webview-tag#preload) 
```
    Object.defineProperty(navigator.serviceWorker, 'register', {
        value: () => {
            return Promise.reject()
        }
    });
```
Hope this helps.

**Comment by IsmaelMartinez** [comment]
Thanks @basaas , I will give it a try once I got some time. If noone else beats me to it!!

**Comment by Lomanic** [comment]
Can confirm @basaas tip fixes the blank page issue (also on Rambox, PR opened), tested with the following diff

```diff
diff --git a/app/browser/index.js b/app/browser/index.js
index 59a8986..98eea85 100644
--- a/app/browser/index.js
+++ b/app/browser/index.js
@@ -88,5 +88,9 @@
                injector.get('settingsService').appConfig.hideGetAppButton = true;
                injector.get('settingsService').appConfig.enableMobileDownloadMailDialog = false;
        }
+       Object.defineProperty(navigator.serviceWorker, 'register', {
+        value: () => {
+            return Promise.reject()
+        }
+    });
 }());
-
```

Thanks a lot @basaas for the catch!

**Comment by IsmaelMartinez** [comment]
Great! Can someone create a PR? (@Lomanic ?) I am happy to merge and send it for test around. Like that then we can probably formally release early next week. Probably reverting the cacheClean, etc that I added as a PoC. 

Basically removing the stuff in the PR https://github.com/IsmaelMartinez/teams-for-linux/pull/215/files. including the 2 webview.webcontent.cache... stuff. 

Thanks a lot @basaas !

**Comment by IsmaelMartinez** [comment]
@inglor have implemented the solution suggested by @basaas and tested by @Lomanic .

Thanks a lot!

**Comment by IsmaelMartinez** [comment]
@skyghis , seems like where also closed to do a PR on this! Thanks to you also.

**Comment by skyghis** [comment]
Yes, I just see it while preparing the PR, I'm too slow :snail:

 Good to see the issue fixed !

**Comment by prazooo** [comment]
So what version is the fix available?

**Comment by IsmaelMartinez** [comment]
just pre-released 0.6.6 so people that don't use snap can test it also. 
https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v0.6.6

The actual release will be either 0.7.0 or 1.0.0... if I get the time to fix the screenshare with electron 6

**Comment by IsmaelMartinez** [comment]
To note that you will need to clean the config folder the 1st time as the user agent changed in 0.6.5 to fix another issue.

**Comment by ghostface** [comment]
Thank you! Works fine now with 0.6.6

**Comment by Maks-Chornyi** [comment]
guys. i'm really sorry, but can't understand. ok. i have the same issue. i see this release:
https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v0.6.6
i downloaded teams-for-linux-0.6.6.tar.gz
i unpacked it. but.. how can i implement this changes,, this release to my team-for-linux ...???

**Comment by IsmaelMartinez** [comment]
What did you install previously? snap, deb, rpm? if so, download a packaged version, it will safe time. 

If using deb (or rpm), you will need to remove the following folder `~/.config/teams-for-linux`

**Comment by Maks-Chornyi** [comment]
> What did you install previously? snap, deb, rpm? if so, download a packaged version, it will safe time.
> 
> If using deb (or rpm), you will need to remove the following folder `~/.config/teams-for-linux`

i used this guide:
https://websiteforstudents.com/install-microsoft-teams-desktop-app-on-ubuntu-16-04-17-10-18-04/
so as i understand i installed snap, i guess
i tried to log in to 'Teams', but had white screen initializing
i checked and found that haven't inside my 'config' folder 'temas-for-linux' folder.
i really understand that these questions are stupid. but just wanna setup this program in my ubuntu 18.04

**Comment by IsmaelMartinez** [comment]
I did make a comment to it as we are not using --beta channel.

If you want to change channel in snap, I think you can use the following command

`sudo snap refresh teams-for-linux --channel=edge`

**Comment by Maks-Chornyi** [comment]
> I did make a comment to it as we are not using --beta channel.
> 
> If you want to change channel in snap, I think you can use the following command
> 
> `sudo snap refresh teams-for-linux --channel=edge`

Thank You!!! it works properly now.

**Comment by mattdconnell** [comment]
Confirmed fixed on 0.6.6 AppImage version as well!

Thank you everyone, you make my work-life a little bit easier.

**Comment by seasurfpete** [comment]
working for me too. thanks

**Comment by mweirauch** [comment]
Thanks everybody involved and @IsmaelMartinez for freeing up some time to ship all the test versions allthough your schedule is tight!

**Comment by alfsch** [comment]
AppImage 0.6.6 works perfect. Big thanx to all :1st_place_medal:

**Comment by joelnibejpi** [comment]
While waiting for the fix to be included in a stable release, I open "Application -> Debug" and then go to "Clear storage" under "Application" tab where I use the "Clear site data" button followed by "Application -> Refresh" to reliably work around the issue. That way, I don't have to remove any files manually and more importantly don't have to log in again.

**Comment by IsmaelMartinez** [comment]
0.6.6 seems to be pretty stable. Hopefully I release it by the end of the week.

**Comment by drozel** [comment]
still have the issue on 0.6.6-496 (installed from the last .deb)

**Comment by IsmaelMartinez** [comment]
Did you do a full config folder clean?

**Comment by drozel** [comment]
@IsmaelMartinez sorry for the stupid question: where is it?

**Comment by IsmaelMartinez** [comment]
Check in the known issues (blank screen) . Sorry but I am not fully online
at the moment

**Comment by drozel** [comment]
@IsmaelMartinez yep, after cleaning the settings it works on 0.6.6. Thanks

**Comment by jamesdwilson** [comment]
Vote to re-open. Asking people to delete config or cache is not a valid solution.

**Comment by IsmaelMartinez** [comment]
@jamesdwilson this is closed as that delete config should only be necessary once. Just after, or before, installing 0.6.6. maybe it wasn't clear.

**Comment by jamesdwilson** [comment]
thank you for that clarification, @IsmaelMartinez  and thank you for your work on this project.

**Comment by paddotk** [comment]
I have a similar issue and the behavior of the app is altogether quite strange. This is what happened:

1. Saw a blank page after hitting the 'refresh' option, quitting and restarting the app didn't work.
2. Quit the app and removed the cache folder as suggested, restarted the app again.
3. It functioned normally at first, but upon refreshing it broke again. No 'Application Cache' folder was created this time.
5. I quit the app again and upgraded (from 0.3.0) to 0.6.0.
6. I was able to login, but then immediately got a blank screen again.
7. I removed the application from Ubuntu Software Center
8. Installed 0.6.0 again, still broken
9. Now the Application Cache folder gets created again and removing it seems to temporarily resolve the issue again, although I don't dare to refresh anymore :p. Strange thing is that in Ubuntu Software Center, the application does not show up anymore (installed it with the .deb file), also in the installations history list.

**Comment by IsmaelMartinez** [comment]
Hi @paddotk , 0.6.6 has the fix. 0.6.0 does **not** have a fix. 

0.6.6 is available either in snap edge channel or in the release folder (marked as "test only" but is actually release ready).

You will need to remove the `~/.config/teams-for-linux` folder after installing 0.6.6 due changes in the user agent (from #204). That should only be needed to be done once.

**Comment by Jalle19** [comment]
> You will need to remove the `~/.config/teams-for-linux` folder after installing 0.6.6 due changes in the user agent (from #204). That should only be needed to be done once.

I can verify that this works

**Comment by paddotk** [comment]
@IsmaelMartinez Awesome! Thanks for your quick response. I didn't try 0.6.6 because of the _test only_ note but if that fixes it I'll just get that one :)

**Comment by phouverneyuff** [comment]
> Also when cleaning config "~/.config/Teams for Linux" it solves the issue for a while, then you will encounter this again.

For me was:
'rm -rf ~/.config/teams-for-linux/Partitions/teams-4-linux/{Application\ Cache,Service\ Worker}/'

**Comment by IsmaelMartinez** [comment]
@phouverneyuff , check version 0.6.6/0.6.7

**Comment by andrewdurkiewicz** [comment]
> Also when cleaning config "~/.config/Teams for Linux" it solves the issue for a while, then you will encounter this again.

This is a dumb question, but by 'clean' what do you mean? Can you give an example please?

**Comment by abelsromero** [comment]
> > Also when cleaning config "~/.config/Teams for Linux" it solves the issue for a while, then you will encounter this again.
> 
> This is a dumb question, but by 'clean' what do you mean? Can you give an example please?

Not dumb at all. He just means deleting the directory `.config/Teams for Linux` or `.config/teams-4-linux` (depending on distro package) located in your HOME directory. This holds several caches, note that after that you will need to log again.

**Comment by IsmaelMartinez** [comment]
No dump questions ;) The config folder location depends on your packaging system:
https://github.com/IsmaelMartinez/teams-for-linux/blob/develop/KNOWN_ISSUES.md#config-folder-locations.

@agates documented that the other day. For snap, you should use snap edge channel. I think is something like this:
 
`sudo snap refresh teams-for-linux --channel=edge`

**Comment by craigs100** [comment]
Getting the blank screen on Linux Mint 18.3 Cinnamon but can't find .config/teams-for-linux/Partitions/teams-4-linux/Application Cache. Anyone any idea where to find the application cache on Mint? Was installed from the LM Software Manager which uses FlatHub.
Thanks.

**Comment by agates** [comment]
@craigs100 as mentioned in the URL in the comment before yours, the configuration directory for a flatpak is:
`~/.var/app/com.github.IsmaelMartinez.teams_for_linux/config/teams-for-linux/`

**Comment by IsmaelMartinez** [comment]
@agates you beat me to this! Thanks for the quick reply. 

0.7.0 is just been build and hopefully I finish the release notes soon, that has this fix. 

Flathub seems to be a bit out of sink, maybe someone can create a PR on it to update to version 0.7.0? The maintainer seems to be offline (probably on holidays)

**Comment by craigs100** [comment]
Thanks @agates. Got it. Sorry to trouble you. FlatHub didn't register with me when I read it for some reason.

**Comment by paragenic** [comment]
Hi, installing on Ubuntu 18.04:

sudo snap install teams-for-linux
teams-for-linux 0.7.0 from Julian Alarcon (alarconj) installed

A JavaScript error occurred in the main process
Uncaught Exception:
/snap/teams-for-linux/56/resources/app.asar/app/config/index.js:91
<<<<<<< HEAD
^^

SyntaxError: Unexpected token <<
    at new Script (vm.js:79:7)
    at createScript (vm.js:251:10)
    at Object.runInThisContext (vm.js:303:10)
    at Module._compile (internal/modules/cjs/loader.js:660:28)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)

**Comment by assembledStarDust** [comment]
Additional: Version 0.7.0 this issue will still exist, but brought on by clicking x to close the window at the end of the previous session. Then this issue will present itself on the following session. Workaround by clearing cache as previously noted works.

This issue can be avoided by logging off cleanly in the previous session before closing the window.

System Setup:
Centos 7
Company Setup: SSO

**Comment by trialism** [comment]
I still have this in the flatpak package.

**Comment by IsmaelMartinez** [comment]
Hi @trialism , have you tried with deb or snap packages?

**Comment by trialism** [comment]
I'm on arch, those won't work for me.

**Comment by IsmaelMartinez** [comment]
try the arch package then https://aur.archlinux.org/packages/teams-for-linux

**Comment by craigs100** [comment]
I've now downloaded the "Microsoft Teams for Linux" Preview package and it seems to be working very well.

**Comment by seboss666** [comment]
It works well, and having native call and calendar notifications is really a must that Ismael's work can't work around (it's the way web version works, inside the window). But CPU usage is still a mess, like it was under Windows and Microsoft did not answer a single word on it for years.

Hence, the preview version may embed more telemetry stuff to monitor key points of this specific build.

**Comment by IsmaelMartinez** [comment]
hi @seboss666, I did look at the asar file and it doesn't look like they have added extra telemetry. Its just the normal telemetry that they have. Is the same code as for Windows and Mac.

You can try to limit the CPU allocation, but then it goes pretty slow if you don't give it 90%

**Comment by magnusolsson80** [comment]
Hi. I'm currently suffering from this problem too in my corporate network. Correct me if I'm wrong, but does the app uses chromium in some way?

I have an issue with chromium that all sites are available except the local authentication server. I just get a error page that the server can't be found. When I try to log into other Microsoft web apps, after the first authentication, the page becomes white. I don't see this issue in Firefox.

I don't know if this is just an issue with Chromium on Ubuntu or not, but my co-worker runs Fedora and he does not see any issues.

**Comment by seboss666** [comment]
> Hi. I'm currently suffering from this problem too in my corporate network. Correct me if I'm wrong, but does the app uses chromium in some way?
> 
> I have an issue with chromium that all sites are available except the local authentication server. I just get a error page that the server can't be found. When I try to log into other Microsoft web apps, after the first authentication, the page becomes white. I don't see this issue in Firefox.
> 
> I don't know if this is just an issue with Chromium on Ubuntu or not, but my co-worker runs Fedora and he does not see any issues.
> 
> --Update: I added our corporate certificate to chromium and now it works.
> --Update 2: I found a work around that at least made it work. By appending "teams-for-linux --ignore-certificate-errors". Happy days!

Electron apps, just like this one, are, in a nutshell, webapps running with a chromium frontend and a nodejs/v8 backend. So yes, it uses Chromium, but a packaged version not dependent on you system dependencies like I think the version shipped on Ubuntu.

**Comment by IsmaelMartinez** [comment]
Hi @magnusolsson80 , your company probably uses self sign certificates.

You will see a message in the console saying something like 
`The unknown certificate fingerprint is:`

if that is the case, you can add that certificate fingerprint to you config.json file.

You can find more info in here https://github.com/IsmaelMartinez/teams-for-linux/tree/develop/app/certificate

Thanks for putting your solution. This helps others to tackle issues and it is really appreciated.

**Comment by jonjahren** [comment]
I met this bug in 1.0 under Ubuntu 19.10, and I tried installing it through flatpak, snap and .deb, but no matter what I did I was still met with no window upon starting the program. I tried the --disable-gpu command line switch, ctrl+r or refresh from context menu, and nothing would work. 

I then tried installing it on my laptop which has an Intel GPU, and it worked out of the box. Then I tried removing one of my GPUs in my desktop, since I was running a GPU passthrough system I thought that might help, but no dice. As a last ditch attempt I tried installing Plasma and when running plasma, the application starts as normal. So my problem exists only under Gnome and only using an AMD GPU.

**Comment by IsmaelMartinez** [comment]
@jonjahren, have you tried something like this? https://github.com/IsmaelMartinez/teams-for-linux/issues/35#issuecomment-474760816

**Comment by mwtoews** [comment]
I just had a similar issue (blank on startup) with the .deb version on Ubuntu (teams 1.3.00.5153 from https://packages.microsoft.com/repos/ms-teams/), which was solved differently than any of the answers above.

My issue was that my laptop was docked and had multiple monitors running. The solution was to un-dock and use the laptop monitor, where I can finally see a log-in screen. After logging-in (and chose option to say logged in), there are no issues when re-docking with multiple monitors, including when restarting.

**Comment by IsmaelMartinez** [comment]
Hi @mwtoews, 

Thanks for reporting that, but you are reporting a MS official app issue. This is NOT the official app. 

I think many people are starting to confuse both... hopefully they solve their main issues soon.

**Comment by manoj-zwayam** [comment]
I quit the app and restarted, it worked for me.

**Comment by CasperOng** [comment]
I believe --no-sandbox does make it working

**Comment by lukasmatt** [comment]
> I believe --no-sandbox does make it working

That finally worked for me. I am happy to contribute some debug logs if you tell me how :)

**Comment by CasperOng** [comment]
@lukasmatt You are welcome. I also want to know how to contribute to some debug logs xD. May I ask you a few questions?

What platform are you on?
What desktop environment or windows manager?
Do you execute teams --no-sandbox on terminal or you use another ways. If yes, how?

**Comment by lukasmatt** [comment]
@CasperOng sure thing :)

I am using Arch Linux with i3 as windows manager. But I think I found the problem.
I am quoting from the AUR package repository (https://aur.archlinux.org/packages/teams):

> Teams is now not compatible with glibc 2.35. It is upstream problem (aka Microsoft has to fix it/update). Here are the workarounds, which can be unstable or insecure. [...]

They suggest using `--no-sandbox` or `--disable-seccomp-filter-sandbox`.

I guess I have to stay put. Downgrading libc is not an option for me.

**Comment by CasperOng** [comment]
> @CasperOng sure thing :)
> 
> I am using Arch Linux with i3 as windows manager. But I think I found the problem.
> I am quoting from the AUR package repository (https://aur.archlinux.org/packages/teams):
> 
> > Teams is now not compatible with glibc 2.35. It is upstream problem (aka Microsoft has to fix it/update). Here are the workarounds, which can be unstable or insecure. [...]
> 
> They suggest using `--no-sandbox` or `--disable-seccomp-filter-sandbox`.
> 
> I guess I have to stay put. Downgrading libc is not an option for me.

Thanks.
I am using Arch Linux with highly modified KDE. I install it with Snap. And so far i have not been facing this problem xDD

But when I first try installing team using AUR. I got this issue thats why I am here btw.

**Comment by ScarletFlash** [comment]
`teams --disable-seccomp-filter-sandbox` worked for me

**Comment by sbow** [comment]
fyi - fixed with "CTRL + R" w whitescreen in focus...

Setup:
- Corporate login w 2 factor auth required & loading of corporate login screen
- CTRL + R made the login screen showup (white previously)
- Linux Mint 21.2
```
❯ cat /etc/*-release
DISTRIB_ID=LinuxMint
DISTRIB_RELEASE=21.2
DISTRIB_CODENAME=victoria
DISTRIB_DESCRIPTION="Linux Mint 21.2 Victoria"
NAME="Linux Mint"
VERSION="21.2 (Victoria)"
ID=linuxmint
ID_LIKE="ubuntu debian"
PRETTY_NAME="Linux Mint 21.2"
VERSION_ID="21.2"
HOME_URL="https://www.linuxmint.com/"
SUPPORT_URL="https://forums.linuxmint.com/"
BUG_REPORT_URL="http://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/"
PRIVACY_POLICY_URL="https://www.linuxmint.com/"
VERSION_CODENAME=victoria
UBUNTU_CODENAME=jammy
cat: /etc/upstream-release: Is a directory
```

**Comment by ipaqmaster** [comment]
On a laptop here of a `12th Gen Intel(R) Core(TM) i7-1255U` CPU which comes with integrated graphics `00:02.0 VGA compatible controller [0300]: Intel Corporation Alder Lake-UP3 GT2 [UHD Graphics] [8086:4628] (rev 0c)`.

No matter how much I reinstalled teams or fully wiped all of its relevant subdirs under my homedir it was starting up with white panels (Which stole focus from every other window during Teams startup) and eventually full-screen white where the SSO screen were supposed to render.

This went away entirely after launching it with `--disable-gpu` then going to settings and disabling graphical acceleration so it persists at startup without needing the flag.

Not sure why teams doesn't like this laptop's onboard i915 graphics.

**Comment by ml31415** [comment]
2025 soon, and this issue still exists.

**Comment by IsmaelMartinez** [comment]
@ml31415 , this is normally a random cache issue when we upgrade electron. Is closed as the clearing of cache solves the problem for 99% of the cases, but some people have been reporting other issues in this thread, what makes it unmanageable. If you got an issue open a new bug/incident and provide as much information as possible, so I can see if I can help you.

**Comment by VladyslavMykhailiuk** [comment]
Maybe, I solve this problem, probably for me, but you can try too : 
1)Change your user agent on your browser to that of Windows. For Firefox change it by visiting about:config.
2) Find the need key
![image](https://github.com/user-attachments/assets/75ebf8f9-e255-4482-90c4-60ddf17742d7)
3) Change this to this -> Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/[113.0.0.0](https://113.0.0.0/) Safari/537.36
